frontend/lib/CMakeLists.txt

```diff
@@ -20,24 +20,26 @@
 # If one exists, the write-git-sha script will only update it if the
 # values are different.
 # If CHPL_DONT_BUILD_SHA is set in the ENV, the git-sha will always be xxxxxxxxxx
-if (EXISTS ${CMAKE_SOURCE_DIR}/.git)
-        execute_process(COMMAND ${CHPL_CMAKE_PYTHON}
-                                ${CMAKE_SOURCE_DIR}/util/config/write-git-sha
-                                ${CMAKE_CURRENT_SOURCE_DIR}/util
-                                --build-version
-                                --chpl-home=${CHPL_HOME})
-        message(STATUS "wrote git-version.cpp")
+find_package(Git)
+if (EXISTS ${CMAKE_SOURCE_DIR}/.git AND Git_FOUND)
+  execute_process(COMMAND ${CHPL_CMAKE_PYTHON}
+                          ${CMAKE_SOURCE_DIR}/util/config/write-git-sha
+                          ${CMAKE_CURRENT_SOURCE_DIR}/util
+                          --build-version
+                          --chpl-home=${CHPL_HOME}
+                          )
+  message(STATUS "wrote git-version.cpp")
 elseif(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/util/git-version.cpp)
-# if we are not in a git repo, but we don't already have a git-version.cpp file,
-# write one as if CHPL_DONT_BUILD_SHA was set so we dont execute the git command
-        execute_process(COMMAND ${CMAKE_COMMAND} -E env
-                        CHPL_DONT_BUILD_SHA=1
-                        ${CHPL_CMAKE_PYTHON}
-                        ${CMAKE_SOURCE_DIR}/util/config/write-git-sha
-                        ${CMAKE_CURRENT_SOURCE_DIR}/util
-        --build-version
-        --chpl-home=${CHPL_HOME})
-        message(STATUS "wrote git-version.cpp")
+# if we are not in a git repo, or the git binary doesn't exist on this machine,
+# write a git-version.cpp file (if one doesn't exist) as if CHPL_DONT_BUILD_SHA
+# was set so we dont try to execute the git command
+  execute_process(COMMAND ${CMAKE_COMMAND} -E env CHPL_DONT_BUILD_SHA=1
+                          ${CHPL_CMAKE_PYTHON}
+                          ${CMAKE_SOURCE_DIR}/util/config/write-git-sha
+                          ${CMAKE_CURRENT_SOURCE_DIR}/util
+                          --build-version
+                          --chpl-home=${CHPL_HOME})
+  message(STATUS "wrote git-version.cpp with dummy xxxxxxxxxx git sha")
```

Comment:
let's leave the message as `wrote git-version.cpp with dummy sha`. The xxxxxxxxxx is arguably too much of an implementation detail.